src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -3117,6 +3117,10 @@ private protected virtual bool SaveAsOnnx(OnnxContext ctx, string[] outputNames,
         }
         bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string featureColumn)
         {
+            const int minimumOpSetVersion = 9;
+            if (ctx.GetOpSetVersion() < minimumOpSetVersion)
+                throw Contracts.ExceptParam(nameof(minimumOpSetVersion), $"Requested OpSet version {ctx.GetOpSetVersion()} is lower than FastTree's minimum OpSet version requirement: {minimumOpSetVersion}");
+
             return SaveAsOnnx(ctx,outputNames,featureColumn);
         }
```

Comment:
The `ISingleCanSaveOnnx.SaveAsOnnx` method here returns by calling the `private protected virtual SaveAsOnnx `method on this class (`TreeEnsembleModelParameters`) which is defined some lines above this method. In turn, this virtual method is overriden in `FastTreeTweedie `and `RandomForestRegression`, where you've also added exceptions checking for the `minimumOpsetVersion`.

Since these SaveAsOnnx methods can each one have a diffirent minimumOpset, I'd suggest moving the exception you 've added here into the `private protected virtual SaveAsOnnx ` method above.